website/pages/docs/deployment/docker-read-only.mdx

```diff
@@ -0,0 +1,122 @@
+
+### Create a config file for the installation
+
+This config file should reference the plugins that you want to install with the actual source where they should be downloaded from:
+
+```yaml
+# install.yaml
+kind: source
+spec:
+  name: xkcd
+  path: hermanschaaf/xkcd
+  version: v2.0.0
+  destinations: ["postgresql"]
+  tables: ["*"]
+---
+kind: destination
+spec:
+  name: "postgresql"
+  registry: "github"
+  path: "cloudquery/postgresql"
+  version: "v6.1.3"
+  spec:
+    connection_string: "postgresql://cq:password123@host.docker.internal:5432/cq-test?sslmode=disable"
+```
+
+### Create a config file for the sync
+
+This config file should reference the plugins running as gRPC servers (fill in the `connection_string`). We will run XKCD on port 7778 and PostgreSQL on port 7779:
+
+```yaml
+# sync.yaml
+kind: source
+spec:
+  name: xkcd
+  registry: grpc
+  path: localhost:7778
+  destinations: ["postgresql"]
+  tables: ["*"]
+---
+kind: destination
+spec:
+  name: "postgresql"
+  registry: grpc
+  path: localhost:7779
+  spec:
+    connection_string: "[REDACTED]"
+```
+
+### Create a start script to start the plugins
+
+This shell script will start the plugins and the sync with the provided parameters when we run the container. To make it easier to maintain, we will use the `find` command to get the path to the actual plugin binary. This way, we don't have to worry about the version of the plugin.
+
+```shell
+#!/bin/sh
+# get path to the xkcd plugin:
+xkcd_plugin_path=$(find .cq -regex ".*/xkcd/.*/plugin")
+# get path to the postgresql plugin:
+postgresql_plugin_path=$(find .cq -regex ".*/postgresql/.*/plugin")
+
+# start the plugins and run the sync with the provided parameters:
+sh -c "${xkcd_plugin_path} serve --address localhost:7778" & \
+sh -c "${postgresql_plugin_path} serve --address localhost:7779" & \
+/app/cloudquery $@
+```
+
+### Connecting it all together
+
+We will use the `ghcr.io/cloudquery/cloudquery:latest` container as a base and add all files in it. Then we will override the entry point with our start script:
+
+```docker
+FROM ghcr.io/cloudquery/cloudquery:latest
+WORKDIR /app
+# this is the install.yaml file we created above
+COPY ./install.yaml /app/install.yaml
+# install the plugins
+RUN /app/cloudquery plugin install install.yaml
+# add the sync config file
+COPY ./sync.yaml /app/sync.yaml
+# add the start script
+COPY --chmod=0755 ./start.sh /app/start.sh
+# override the start script
+ENTRYPOINT [ "/app/start.sh"]
+```
+
+
+## Run the Container
+
+First, build the container as you would normally do:
+
+```bash
+docker build ./ -t my-cq-container:latest
+```
+
+Run the container with `--read-only` option and a command to pass to our start script. You will also need to add the `--no-log-file` option to skip logging to a file. Here is an example:
```

Comment:
I would suggest rather using `--log-console` so that you can still access the logs at least

Reply:
actually, `--no-log-file` is still required. I also didn't see a difference in the log when used the `--log-console`

Reply:
Ah okay yeah seems like `--log-console` results in logs being written to both file and console. So in that case I'd suggest using both: `--log-console --no-log-file` so that the output would only go to stdout/stderr